### PR TITLE
feat(cli): support apply out of sync flag only

### PR DESCRIPTION
**Pull request by pasha-codefresh**

### CODE REVIEW

**Comment by jannfis** [comment]
@pasha-codefresh The code doesn't seem to compile, PTAL.

**Comment by pasha-codefresh** [comment]
@jannfis , yes, sorry, moving it to draft. I am working on e2e things and code completion

**Comment by pasha-codefresh** [comment]
Depend on this PR 
https://github.com/argoproj/gitops-engine/pull/533

**Comment by crenshaw-dev** [comment]
@pasha-codefresh merged the gitops-engine PR

**Comment by pasha-codefresh** [comment]
@jannfis @crenshaw-dev this one is ready for review

**Comment by crenshaw-dev** [comment]
@pasha-codefresh looks like it needs one docs tweak (i.e. CLI codegen).

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -1813,6 +1817,7 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	command.Flags().BoolVar(&force, "force", false, "Use a force apply")
 	command.Flags().BoolVar(&replace, "replace", false, "Use a kubectl create/replace instead apply")
 	command.Flags().BoolVar(&serverSideApply, "server-side", false, "Use server-side apply while syncing the application")
+	command.Flags().BoolVar(&applyOutOfSyncOnly, "apply-out-of-sync-only", false, "Use out-of-sync apply that sync only out of sync resources")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:1820]:
```suggestion
	command.Flags().BoolVar(&applyOutOfSyncOnly, "apply-out-of-sync-only", false, "Sync only out-of-sync resources")
```

I'd simplify the language just a bit.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by pasha-codefresh** [comment]
@crenshaw-dev could you please take a look?

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks Pasha!

**Pull request merged**